rust/core-lib/src/editor.rs

```diff
@@ -1019,4 +1021,30 @@ mod tests {
 
         assert_eq!(editor.get_buffer().to_string(), "sshello");
     }
+
+    #[test]
+    fn empty_transpose() {
```

Comment:
these tests should probably go in `event_context.rs`; there's a nice harness there for testing these kinds of commands.